cluster/calcium/status.go

```diff
@@ -22,23 +23,36 @@ func (c *Calcium) GetWorkloadsStatus(ctx context.Context, ids []string) ([]*type
 }
 
 // SetWorkloadsStatus set workloads status
-func (c *Calcium) SetWorkloadsStatus(ctx context.Context, status []*types.StatusMeta, ttls map[string]int64) ([]*types.StatusMeta, error) {
-	logger := log.WithField("Calcium", "SetWorkloadsStatus").WithField("status", status[0]).WithField("ttls", ttls)
+func (c *Calcium) SetWorkloadsStatus(ctx context.Context, statusMetas []*types.StatusMeta, ttls map[string]int64) ([]*types.StatusMeta, error) {
+	logger := log.WithField("Calcium", "SetWorkloadsStatus").WithField("status", statusMetas[0]).WithField("ttls", ttls)
 	r := []*types.StatusMeta{}
-	for _, workloadStatus := range status {
-		workload, err := c.store.GetWorkload(ctx, workloadStatus.ID)
-		if err != nil {
-			return nil, logger.Err(ctx, errors.WithStack(err))
+	for _, statusMeta := range statusMetas {
+		// In order to compat
+		if statusMeta.Appname == "" || statusMeta.Nodename == "" || statusMeta.Entrypoint == "" {
+			workload, err := c.store.GetWorkload(ctx, statusMeta.ID)
+			if err != nil {
+				return nil, logger.Err(ctx, errors.WithStack(err))
+			}
+
+			appname, entrypoint, _, err := utils.ParseWorkloadName(workload.Name)
+			if err != nil {
+				return nil, logger.Err(ctx, errors.WithStack(err))
+			}
+
+			statusMeta.Appname = appname
+			statusMeta.Nodename = workload.Nodename
+			statusMeta.Entrypoint = entrypoint
 		}
-		ttl, ok := ttls[workloadStatus.ID]
+
+		ttl, ok := ttls[statusMeta.ID]
```

Comment:
there's a concern here, if no `ttl` found, should `0` be used? or should we skip this status?

Reply:
0 is default value